cam.go

```diff
@@ -181,47 +182,37 @@ func (fc *filteredCamera) Images(ctx context.Context) ([]camera.NamedImage, reso
 	return nil, meta, data.ErrNoCaptureToStore
 }
 
-func (fc *filteredCamera) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
-	camStream, err := fc.cam.Stream(ctx, errHandlers...)
+func (fc *filteredCamera) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+
+	bytes, meta, err := fc.cam.Image(ctx, mimeType, extra)
 	if err != nil {
-		return nil, err
+		return nil, meta, err
 	}
-
-	return filterStream{camStream, fc}, nil
-}
-
-type filterStream struct {
-	cameraStream gostream.VideoStream
-	fc           *filteredCamera
-}
-
-func (fs filterStream) Next(ctx context.Context) (image.Image, func(), error) {
-	extra, ok := camera.FromContext(ctx)
-	if !ok || extra[data.FromDMString] != true {
-		// If not data management collector, return underlying stream contents without filtering.
-		return fs.cameraStream.Next(ctx)
+	ex, ok := ctx.Value(0).(map[string]interface{})
+	if !ok || ex[data.FromDMString] != true {
+		// If not data management collector, return underlying contents without filtering.
+		return bytes, meta, err
 	}
 
-	img, release, err := fs.cameraStream.Next(ctx)
+	img, err := rimage.DecodeImage(ctx, bytes, mimeType)
```

Comment:
did you know that the mimeType you used in the `Image` request may not be the mimeType you get back? (this is a fun quirk of our camera proto). Instead you should use extract the mimeType returned in the `meta` variable and use that one. 

Make a unit test of a fake camera returning a mimeType that is not the same as the mimeType requested, and see if GetImage still succeeds.